### DISCUSSION TITLE
Unable to find data in Response exhaust report

### DISCUSSION THREAD

**Discussion by sravani-64** [Sunbird-Lern/Community#144] [Issues] [answered]
Hi team,

We have setup reports 4.10 version. We are able to submit progress exhaust and response exhaust jobs from portal UI and triggered the cron job manually for both reports. We are able to see that progress exhaust report is generated with data and downloaded from portal UI as well. 
But for response exhaust, report file is generated but could not see any data in the report. To verify this, we have performed following analysis.
1. Verified if the job configurations is pointing to right Redis, Elastic search and Cassandra DBs and found that the configurations are correct.
2. Verified if we have relevant data in Redis and Cassandra as per https://lern.sunbird.org/learn/product-and-developer-guide/lms-service/reports/on-demand-exhaust/response-exhaust 
3. We were able to find the data in **user_enrolments**
![image](https://github.com/Sunbird-Lern/Community/assets/133327173/7a53d71b-38a7-48b9-b2f5-d6c8ec1eb46f)
**user_activity_agg**
![image](https://github.com/Sunbird-Lern/Community/assets/133327173/335d2933-ba2c-41a7-a48b-4faf47e98125)
but no data was found in **assessment_aggregator**
![image](https://github.com/Sunbird-Lern/Community/assets/133327173/36fb29eb-74cb-44ae-aa10-a9c808c1d31f)

Could someone please help us to understand what might be causing this issue? Thank you

**Comment by Jayaprakash8887** [comment]
@sravani-64 Report exhaust data are compiled using Obsrv. Kindly raise a thread in Sunbird Obsrv as well. 

@kumarks1122 Please check this thread.

**Reply by sravani-64** [reply]
@Jayaprakash8887  Sure. Will add it in Obsrv also

**Reply by kumarks1122** [reply]
@sravani-64 : Response exhaust report contains the response data of user for an assessment or questionset. Can you please confirm, if the course is having any assessments or questionsets?
cc: @Jayaprakash8887

**Reply by sravani-64** [reply]
@kumarks1122  The course is having questionsets. After submitting request for questionset from UI, I have triggered Response-exhaust cronjob manually. Submit request is successful but the report generated has no data

**Reply by kumarks1122** [reply]
@sravani-64 : If course has questionsets, user's response will be recorded in assessment aggregator table through assessment aggregator flink job. Kindly check, if that flink job is running fine and it is pointing correct cassandra?

**Comment by sravani-64** [comment]
@kumarks1122 yes, assessment aggregator flink job is running fine and pointing to correct Cassandra

**Reply by kumarks1122** [reply]
@sravani-64 Can you check if the events are available in assessment aggregator flink job's input topic? 
If not, kindly check if the course service is configured with correct kafka cluster and kafka topic same as the flink job has?

**Reply by Nithin-ns24** [reply]
@kumarks1122 
events are available in assesment-aggregator flink-jobs input topic and kafka ip is correctly configured
![image](https://github.com/Sunbird-Lern/Community/assets/99644046/df777e23-6daf-4503-848a-5930a4c194e8)

**Reply by sravani-64** [reply]
@kumarks1122 It has the same input topic also in course i.e., lms service configmaps
![image](https://github.com/Sunbird-Lern/Community/assets/133327173/bb9937e2-de49-4b2b-b7a4-afa6c803fa14)

**Reply by sravani-64** [reply]
@kumarks1122 I have checked and confirmed that events are also present in Kafka machine
![image](https://github.com/Sunbird-Lern/Community/assets/133327173/6d775fc9-3db4-41b7-9f3d-2298f4e5e2ec)

**Reply by kumarks1122** [reply]
@sravani-64 could you please check and confirm whether the cluster configuration is same in flink job and course service?

**Reply by sravani-64** [reply]
@kumarks1122 yes the configurations are same in both flink and lms services

**Reply by kumarks1122** [reply]
@sravani-64 have you checked if the events are pushed into the kafka topic immediately after consuming the course questionset?
If you get the events, can you share?
Since assessment aggregator table is not updated and user_activity_agg table is not having score details, it seems to be issue with processing the events.

**Reply by sravani-64** [reply]
@kumarks1122 
![image](https://github.com/Sunbird-Lern/Community/assets/133327173/d3e7008d-4541-43d6-b10f-e84d4ae0a4f7)
These are the events from Kafka

**Reply by Jayaprakash8887** [reply]
@kumarks1122 Please update the summary of the discussion/solution

**Reply by kumarks1122** [reply, marked as answer]
@sravani-64  I've included for you the summary of our discussion offline.
Course service is directly pushing the events to assess events Kafka topic. Also, we didn't find the `ASSESS` events related to that expected course in that Kafka topic.
So decided to check the event generation in the portal. We found out that the portal is not sending the assess events to the course service. 
So, we decided to check with the portal team for the event generation issue.
cc: @Jayaprakash8887 @AmiableAnil
